src/components/MessageItem/index.tsx

```diff
@@ -0,0 +1,186 @@
+      return null;
+    }
+
+    return avatar?.fileUrl ? (
+      <Avatar.Image source={{ uri: avatar?.fileUrl }} style={messageStyles.avatar} size={40} />
+    ) : (
+      <Avatar.Text
+        size={40}
+        style={messageStyles.avatar}
+        label={user?.displayName?.slice(0, 2).toUpperCase() ?? ''}
+      />
+    );
+  }, [avatar, isSameThread, user]);
+
+  const MessageHeader = useCallback(() => {
+    return isSameThread ? null : (
+      <View style={messageStyles.headerView}>
+        <Text
+          style={[messageStyles.standardFont, messageStyles.headerItem, messageStyles.username]}
+        >
+          {user?.displayName}
+        </Text>
+      </View>
+    );
+  }, [isSameThread, user]);
+
+  const Day = useCallback(() => {
+    return isSameDay ? null : (
+      <View style={messageStyles.timeContainerStyle}>
+        <Text style={messageStyles.timeTextStyle}>
+          {dayjs(messageProp.createdAt).format('YYYY MMMM DD')}
+        </Text>
+      </View>
+    );
+  }, [isSameDay, messageProp]);
+
+  const RenderMessageImage = () => {
+    if (type === 'image' && messageData) {
+      return <MessageImage message={messageData} />;
+    }
+    return null;
+  };
+
+  const RenderMessageText = () => {
+    if (type === 'text' && messageData) {
+      return (
+        <MessageText
+          message={messageData}
+          textStyle={{ color: colors.text, fontFamily: regularFont }}
+        />
+      );
+    }
+    return null;
+  };
+
+  const container = useMemo(() => ({ marginTop: isSameUser ? 2 : 20 }), [isSameUser]);
+  const containerStyle = useMemo(
+    () => [messageStyles.container, currentUserId === userId && messageStyles.containerRight],
+    [currentUserId, userId],
+  );
+  const messageHeaderStyle = useMemo(
+    () => [
+      messageStyles.messageHeader,
+      currentUserId === userId && messageStyles.messageHeaderRight,
+    ],
+    [currentUserId, userId],
+  );
+  const pressableStyle = useMemo(
+    () => [
+      messageStyles.bubbleWrapper,
+      currentUserId === userId && messageStyles.bubbleWrapperRight,
+      currentUserId === userId && { backgroundColor: colors.primary },
+      currentUserId !== userId && { backgroundColor: colors.accent },
+    ],
+    [colors.accent, colors.primary, currentUserId, userId],
+  );
+  const leftArrowStyle = useMemo(
+    () => [
+      messageStyles.leftArrow,
+      currentUserId === userId && messageStyles.rightArrow,
+      currentUserId === userId && { backgroundColor: colors.primary },
+      currentUserId !== userId && { backgroundColor: colors.accent },
+    ],
+    [colors.accent, colors.primary, currentUserId, userId],
+  );
+  const leftArrowOverlapStyle = useMemo(
+    () => [
+      messageStyles.leftArrowOverlap,
+      currentUserId === userId && messageStyles.rightArrowOverlap,
+      {
+        backgroundColor: dark ? overlay(4, colors.surface) : colors.surface,
+      },
+    ],
+    [colors.surface, currentUserId, dark, userId],
+  );
```

Comment:
if you have a separate style file, let's move all the logic there as much as possible. export fns with parameters if necessary for clean separate of concerns

Reply:
honestly I was thinking on this case of 1-2 days, as in old android the performance is an issue, it should be optimised, also clean as you said
also we have some style properties in context! so anyway we should pass them
in this case I ended up using zacs, something near styled-components
would you confirm its using?